x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/calculations/counter_rate.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { FormattedIndexPatternColumn, ReferenceBasedIndexPatternColumn } from '../column_types';
+import { IndexPatternLayer } from '../../../types';
+import { checkForDateHistogram, dateBasedOperationToExpression, hasDateField } from './utils';
+import { OperationDefinition } from '..';
+
+const ofName = (name: string) => {
+  return i18n.translate('xpack.lens.indexPattern.CounterRateOf', {
+    defaultMessage: 'Counter rate of {name}',
+    values: { name },
+  });
+};
+
+export type CounterRateIndexPatternColumn = FormattedIndexPatternColumn &
+  ReferenceBasedIndexPatternColumn & {
+    operationType: 'counter_rate';
+  };
+
+export const counterRateOperation: OperationDefinition<
+  CounterRateIndexPatternColumn,
+  'fullReference'
+> = {
+  type: 'counter_rate',
+  priority: 2,
+  displayName: i18n.translate('xpack.lens.indexPattern.counterRate', {
+    defaultMessage: 'Counter rate',
+  }),
+  input: 'fullReference',
+  selectionStyle: 'field',
+  requiredReferences: [
+    {
+      input: ['field'],
+      specificOperations: ['max'],
+      validateMetadata: (metadata) =>
+        metadata.dataType === 'number' && metadata.isBucketed === false,
```

Comment:
Very minor comment, but I've been trying to inline these.

```suggestion
      validateMetadata: (meta) => meta.dataType === 'number' && !metadata.isBucketed,
```